Lib/json/tool.py

```diff
@@ -25,24 +25,35 @@ def main():
                         help='a JSON file to be validated or pretty-printed')
     parser.add_argument('outfile', nargs='?', type=argparse.FileType('w'),
                         help='write the output of infile to outfile')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--indent', default=4, type=int,
+                       help='Indent level (number of spaces) for '
+                       'pretty-printing. Defaults to 4.')
+    group.add_argument('--no-indent', action='store_const', dest='indent',
```

Comment:
@haypo I agree the most compact representation would be useful.

One option would be add a third argument (`--compact`) to the mutually exclusive group. This would provide access to three options:

1. `--indent`: pretty (with newlines)
2. `--no-indent`: default `json.dump`
3. `--compact`: compact, with `separators=(',', ':')` in `json.dump`

Alternatively, are you proposing to only support options 1 & 3?

I'm happy to implement this. Let's get a second opinion, so I don't go down the wrong track. Tagging @serhiy-storchaka, @berkerpeksag, @methane.